src/err/mod.rs

```diff
@@ -322,6 +321,31 @@ impl PyErr {
         }))
     }
 
+    #[cfg(Py_3_12)]
+    fn _take(py: Python<'_>) -> Option<PyErr> {
+        let pvalue = unsafe { Py::from_owned_ptr_or_opt(py, ffi::PyErr_GetRaisedException()) }?;
+        let state = PyErrStateNormalized { pvalue };
+        if state.ptype(py).as_ptr() == PanicException::type_object_raw(py).cast() {
+            let msg: String = state.pvalue.as_ref(py).to_string();
```

Comment:
I think the same nit w.r.t. `.str().ok()` instead of calling into `Display` applies here.